casts/cards/src/UICollectionCard/CollectionSlide.tsx

```diff
@@ -4,30 +4,57 @@ import { UIImage, UIVideo } from '@tonlabs/uikit.media';
 
 import type { CollectionSlideProps } from './types';
 
-function VideoSlide({ content, style }: CollectionSlideProps) {
+function VideoSlide({ content, style, onLoad, isVisible, onError }: CollectionSlideProps) {
     if (!content.source.uri) {
         return null;
     }
 
     return (
         <View style={[StyleSheet.absoluteFill, style]}>
-            <UIVideo uri={content.source.uri} muted repeat resizeMode="cover" />
+            <View
+                style={[
+                    StyleSheet.absoluteFill,
+                    {
+                        opacity: isVisible ? 1 : 0,
+                    },
+                ]}
+            >
+                <UIVideo
+                    uri={content.source.uri}
+                    muted
+                    repeat
+                    resizeMode="cover"
+                    onLoad={onLoad}
+                    onError={onError}
+                />
+            </View>
         </View>
     );
 }
 
 function CollectionSlideImpl(props: CollectionSlideProps) {
-    const { content, style } = props;
+    const { content, style, onLoad, isVisible, onError } = props;
     switch (content.contentType) {
         case 'Image':
             return (
                 <View style={[StyleSheet.absoluteFill, style]}>
-                    <UIImage source={content.source} style={StyleSheet.absoluteFill} />
+                    <UIImage
+                        source={content.source}
+                        style={[
+                            StyleSheet.absoluteFill,
+                            {
+                                opacity: isVisible ? 1 : 0,
+                            },
+                        ]}
+                        onLoad={onLoad}
+                        onError={onError}
+                    />
                 </View>
             );
         case 'Video':
             return <VideoSlide {...props} />;
         default:
+            setTimeout(() => onError());
```

Comment:
Could you leave a comment why you wrapped it with `setTimeout`? Also the whole construction looks weird during `render`, can you explain why so? 🤔

Reply:
I agree. Fixed